oe_theme.theme

```diff
@@ -1522,3 +1523,24 @@ function oe_theme_preprocess_page(array &$variables): void {
     'id' => ['main-content'],
   ]);
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_pattern_organisation_teaser(array &$variables): void {
+  $contact_page_url = $variables['contact_page_url'];
+
+  // Define icon based on the url type.
+  $variables['contact_page']['url'] = $contact_page_url;
+  if (UrlHelper::isExternal($contact_page_url)) {
+    $variables['contact_page']['icon'] = 'external';
+    $variables['contact_page']['url'] = $contact_page_url;
+  }
```

Comment:
The problem here is very specific to how UI Patterns are processed. So basically, patterns receive their theme suggestions based on the context where they are rendered. Because of this more specific suggestions will be executed after this one, which means that more specific suggestions will not have the chance to set the `$variables['contact_page_url']` field soon enough for this code to be ran.

This is the reason why we process everything in the templates, so that we can be sure that that happens late enough for us to act and check things like `UrlHelper::isExternal()`.

I'd then quickly create a Twig filter for `UrlHelper::isExternal()` in our `\Drupal\oe_theme_helper\TwigExtension\TwigExtension` class, so that we can move all of this right in the pattern template, effectively removing this whole preprocessing.

Since that's a static method and twig filters accept a callable, we could simply use:

```php
  /**
   * {@inheritdoc}
   */
  public function getFilters(): array {
    return [
      new \Twig_SimpleFilter('is_external_link', [UrlHelper::class, 'isExternal']),
```